src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,45 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_MAX_SIZE = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[])new Object[STORAGE_MAX_SIZE];
+        values = (V[])new Object[STORAGE_MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (compareKeys(keys[i], key)) {
+                values[i] = value;
+                return;
+            }
+        }
+        keys[size] = key;
+        values[size] = value;
+        size++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
+            if (compareKeys(keys[i], key)) {
+                return values[i];
+            }
+        }
         return null;
     }
 
+    public boolean compareKeys(Object key1, Object key2) {
+        return (key1 == key2 || key1 != null && key1.equals(key2));
+    }
```

Comment:
Let's make keys parameterized and remove redundant parentheses
```suggestion
    public boolean compareKeys(K key1, K key2) {
        return key1 == key2 || key1 != null && key1.equals(key2);
    }
```